src/embedding.rs

```diff
@@ -191,6 +277,88 @@ impl MatrixWrapper for MMapMatrix {
         }
     }
 
+    fn init_with_evec(
```

Comment:
In rust you can easily inline unit tests into code.

I suggest you make such a test with eigenvalue initialization where you take some predefined vectors, do the eigenvalues initialization and check if output is as expected.

You can event take input/outputs from your python development code to make sure that rust implementation works as expected.